### DISCUSSION TITLE
Gallery of revealjs presentations with title slide images

### DISCUSSION THREAD

**Discussion by nurfatimaj** [quarto-dev/quarto-cli#9029] [Q&A]
### Description

Hi!

I would like to include a listing of revealjs presentations on my website such that title slides are used as card images. So, it would look similar to the embedded presentation here: https://quarto.org/docs/presentations/revealjs/, but just a simple image. 

I have tried to include a gallery listing, but that just returns empty boxes.
![image](https://github.com/quarto-dev/quarto-cli/assets/24235395/c8e6234e-542f-4ad9-82e5-d0234d33d04a)

I have tried to create a custom listing where the template `gallery.ejs` uses the embedding code for card image:

```{=html}
<div class="list grid quarto-listing-cols-2">
  <% for (const item of items) { %>
    <div class="card border-2 g-col-1" <%= metadataAttrs(item) %>>
      <div class="card-header">
        <p class="listing-title"><%- item.title %></p>
        <div class="card-img-top"><iframe class="slide-deck" src="<%- item.path %>"></iframe></div>
      </div>
    </div>
  <% } %>
</div>
```

But instead of having embedded slide deck, the card image just shows the content of the .qmd file starting from YAML front matter.  
![image](https://github.com/quarto-dev/quarto-cli/assets/24235395/10f64e46-88c2-4886-aa00-8215fb8bcb0b)

I also tried using `<%- item.outputHref %>` or `<%- item.href %>` instead of `<%- item.path %>`. When I use `<%- item.outputHref %>` I only see "Not Found" instead of the slide preview. I think this happens because `<%- item.outputHref %>` evaluates to `teaching/course/lecture-1/lecture.html`. The listing is in the page `teaching/course/index.html`, so I think the embed tries to point to `teaching/course/teaching/course/lecture-1/lecture.html`, which doesn't exist => "Not Found" message. When I use `<%- item.href %>`, it's just empty because `item` doesn't have field with name `href` (I confirmed it by printing all available fields using `<%- Object.keys(item) %>`).

Is there a nice and easy way of including a gallery listing where first slide images are automatically used as preview images? I know that I can take screenshots of title slides and add them as featured image that the gallery listing should be able to use as card image.

**Comment by mcanouil** [comment]
A Reveal.js document is not an image. If you want an image, you need to create one, using tools such as [`decktape`](https://github.com/astefanutti/decktape) or `webshot2` in R.

Also see the exact same question already discussed:
- https://github.com/quarto-dev/quarto-cli/discussions/1840
